site/docs/guides/keycloak.md

```diff
@@ -0,0 +1,72 @@
+# Authentication with Keycloak
+
+In this guide we walk through the process of configuring a Nessie Server to authenticate clients against 
+a local [Keycloak](https://www.keycloak.org/) server. Docker is use at the runtime environments for both servers.
+
+## Setting up Keycloak
+
+For the purposes of this guide we will only do use a simple Keycloak configuration, that is still sufficient to
+demonstrate how OpenID authentication works in Nessie servers.
+
+First, start a Keycloak container using its latest Docker image.
+
+```shell
+docker run -p 8080:8080 -e KEYCLOAK_USER=admin -e KEYCLOAK_PASSWORD=admin \
+  --name keycloak quay.io/keycloak/keycloak:latest
+```
+
+Note the `admin` username and password. Those values will be required to log into the Keycloak Administration Console
+that should now be available at http://localhost:8080/auth/admin/
+
+The default realm is called `Master`. On the left-hand pane find the `Manage > User` page and click `Add User` on the
+right side of the (initially empty) users table.
+
+Enter the username "nessie" and click `Save`. Now, under the `Credentials` tab of the `nessie` user page set password
+to `nessie` and turn off the `Temporary` flag. Click `Set Password`.
+
+For the sake of convenience let's increase the default token expiration time.
+Goto `Clients` > `admin-cli` > `Advanced Settings`. Set `Access Token Lifespan` to 1 day and click `Save`.
+
+Now we are ready to generate an OpenID access token to the `nessie` user:
+
+```shell
+curl -X POST \
```

Comment:
Maybe 
```
export NESSIE_AUTH_TOKEN=$(
    <that curl command> 
)
```

Reply:
I thought about that but I did not want to the example to be bash-specific ;)

Reply:
WDYT about the latest variant? 906af61764b607fd71205d1708deec3f6f638766

Reply:
:+1: